src/Raven.Server/ServerWide/ServerStore.cs

```diff
@@ -1976,7 +1976,9 @@ public Task<(long Index, object Result)> ToggleTaskState(long taskId, string tas
                     if (taskName == null)
                     {
                         if (_server.ServerStore.DatabasesLandlord.DatabasesCache.TryGetValue(dbName, out _) == false)
-                            throw new DatabaseDoesNotExistException($"Can't get subscription name because The database {dbName} does not exists");
+                            if(_server.ServerStore.DatabasesLandlord.ShardedDatabasesCache.TryGetValue(dbName, out _) == false)
+                                throw new DatabaseDoesNotExistException($"Can't get subscription name because The database {dbName} does not exists");
```

Comment:
I think this is a perfect occasion to use (DocumentDatabase || ShardedDatabaseContext).Subscriptions.GetSubscriptionNameById and get rid of pragma few lines below

Reply:
Did something a bit different because this approach caused the compiler to complain about possible NREs etc

Reply:
Implemented Karmel's suggestion for abstract `GetSubscriptionStorage()`